src/main/java/com/mongodb/kafka/connect/sink/cdc/mongodb/operations/Update.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Original Work: Apache License, Version 2.0, Copyright 2017 Hans-Peter Grahsl.
+ */
+
+package com.mongodb.kafka.connect.sink.cdc.mongodb.operations;
+
+import static com.mongodb.kafka.connect.sink.cdc.mongodb.operations.OperationHelper.getDocumentKey;
+import static com.mongodb.kafka.connect.sink.cdc.mongodb.operations.OperationHelper.getFullDocument;
+import static com.mongodb.kafka.connect.sink.cdc.mongodb.operations.OperationHelper.getTruncatedArrays;
+import static com.mongodb.kafka.connect.sink.cdc.mongodb.operations.OperationHelper.getUpdateDescription;
+import static com.mongodb.kafka.connect.sink.cdc.mongodb.operations.OperationHelper.getUpdateDocument;
+import static com.mongodb.kafka.connect.sink.cdc.mongodb.operations.OperationHelper.hasFullDocument;
+import static com.mongodb.kafka.connect.sink.cdc.mongodb.operations.OperationHelper.hasTruncatedArrays;
+import static java.util.Collections.singletonList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.kafka.connect.errors.DataException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.bson.BsonDocument;
+
+import com.mongodb.client.model.ReplaceOneModel;
+import com.mongodb.client.model.UpdateOneModel;
+import com.mongodb.client.model.WriteModel;
+
+import com.mongodb.kafka.connect.sink.cdc.mongodb.ChangeStreamOperation;
+import com.mongodb.kafka.connect.sink.converter.SinkDocument;
+
+public class Update implements ChangeStreamOperation {
+  private static final Logger LOGGER = LoggerFactory.getLogger(Update.class);
+
+  @Override
+  public List<WriteModel<BsonDocument>> perform(final SinkDocument doc) {
+    BsonDocument changeStreamDocument =
+        doc.getValueDoc()
+            .orElseThrow(
+                () ->
+                    new DataException("Error: value doc must not be missing for update operation"));
+
+    BsonDocument documentKey = getDocumentKey(changeStreamDocument);
+
+    if (hasFullDocument(changeStreamDocument)) {
```

Comment:
If the fullDocument is available that is used rather than creating update operation(s)

Reply:
How come?  Using update operations is often more efficient.

Reply:
Is it more efficient?

Reply:
Happy to remove and just use updates, although that will lead to silent errors if users upgrade MongoDB but dont update the connector. Where as fullDocument is guaranteed to work in both scenarios.

Reply:
Can you explain further the update scenario that will lead to silent errors?  I think I'm missing something.

Reply:
So in the future if a user upgrades their MongoDB but not their Kafka connector. If we only use the `updateDescription` and not `fullDocument` then there would be no handling for updates that change array sizes, they would silently be missed.

So I think the choices are either:

1) Use fullDocument if it exists (i.e the user has configured it in the change stream) - this will work for all scenarios
2) Only issue updates - this will work for 4.4 but fail on newer versions

I'll add some checking code so that a hard exception is thrown if there are unknown fields in the `updateDescription` document - that way there will be no silent errors.

Reply:
What did you decide to do?

Reply:
I think we'll keep full support for fullDocument and if a change stream is created without it, then it will error when unexpected fields are present in the `updateDescription`

Reply:
Is the default to create the change stream with fullDocument enabled? My concern is that if it is then by default updates will be slower due to full document replacement.

Reply:
Its off by default.